From 3ef3b17d582bf808b1ac99bd8344b9b33b4cb390 Mon Sep 17 00:00:00 2001
From: ayaka <ayaka@soulik.info>
Date: Wed, 13 May 2015 18:40:55 +0800
Subject: [PATCH 08/12] [v4l2]: add handler of Controls ID

The exist implement could handle control id well, but in the
properties of v4l2 encoder and decoder object, it is ignored.
---
 sys/v4l2/gstv4l2h264enc.c  | 14 --------------
 sys/v4l2/gstv4l2videodec.c |  8 ++++++++
 sys/v4l2/gstv4l2videoenc.c |  8 ++++++++
 3 files changed, 16 insertions(+), 14 deletions(-)

diff --git a/sys/v4l2/gstv4l2h264enc.c b/sys/v4l2/gstv4l2h264enc.c
index 9172146..480c688 100644
--- a/sys/v4l2/gstv4l2h264enc.c
+++ b/sys/v4l2/gstv4l2h264enc.c
@@ -52,26 +52,12 @@ static void
 gst_v4l2_h264_enc_set_property (GObject * object,
     guint prop_id, const GValue * value, GParamSpec * pspec)
 {
-  GstV4l2H264Enc *self = GST_V4L2_H264_ENC (object);
-
-  switch (prop_id) {
-      /* By default, only set on output */
-    default:
-      break;
-  }
 }
 
 static void
 gst_v4l2_h264_enc_get_property (GObject * object,
     guint prop_id, GValue * value, GParamSpec * pspec)
 {
-  GstV4l2H264Enc *self = GST_V4L2_H264_ENC (object);
-
-  switch (prop_id) {
-      /* By default read from output */
-    default:
-      break;
-  }
 }
 
 static GstFlowReturn
diff --git a/sys/v4l2/gstv4l2videodec.c b/sys/v4l2/gstv4l2videodec.c
index 8da5b88..937763c 100644
--- a/sys/v4l2/gstv4l2videodec.c
+++ b/sys/v4l2/gstv4l2videodec.c
@@ -72,6 +72,10 @@ gst_v4l2_video_dec_set_property (GObject * object,
       gst_v4l2_object_set_property_helper (self->v4l2capture, prop_id, value,
           pspec);
       break;
+    case PROP_EXTRA_CONTROLS:
+      gst_v4l2_object_set_property_helper (self->v4l2output, prop_id, value,
+          pspec);
+      break;
 
       /* By default, only set on output */
     default:
@@ -98,6 +102,10 @@ gst_v4l2_video_dec_get_property (GObject * object,
       gst_v4l2_object_get_property_helper (self->v4l2output, PROP_IO_MODE,
           value, pspec);
       break;
+    case PROP_EXTRA_CONTROLS:
+      gst_v4l2_object_get_property_helper (self->v4l2output, prop_id, value,
+          pspec);
+      break;
 
       /* By default read from output */
     default:
diff --git a/sys/v4l2/gstv4l2videoenc.c b/sys/v4l2/gstv4l2videoenc.c
index 0478289..7f8cc56 100644
--- a/sys/v4l2/gstv4l2videoenc.c
+++ b/sys/v4l2/gstv4l2videoenc.c
@@ -76,6 +76,10 @@ gst_v4l2_video_enc_set_property (GObject * object,
       gst_v4l2_object_set_property_helper (self->v4l2capture,
           prop_id, value, pspec);
       break;
+    case PROP_EXTRA_CONTROLS:
+      gst_v4l2_object_set_property_helper (self->v4l2output,
+          prop_id, value, pspec);
+      break;
 
       /* By default, only set on output */
     default:
@@ -103,6 +107,10 @@ gst_v4l2_video_enc_get_property (GObject * object,
           PROP_IO_MODE, value, pspec);
       break;
 
+    case PROP_EXTRA_CONTROLS:
+      gst_v4l2_object_get_property_helper (self->v4l2output,
+          prop_id, value, pspec);
+      break;
       /* By default read from output */
     default:
       if (!gst_v4l2_object_get_property_helper (self->v4l2output,
-- 
2.5.0

